src/Event.hpp

```diff
@@ -77,6 +123,9 @@ class Event {
     // Send time
     unsigned int send_time_ = 0;
 
+    // for comparison
+    std::bitset<bitset_size> comparison_set;
+
     // For differentiating same events which is caused by
     //  anti-message + regeneration of event.
     unsigned long long generation_ = 0;
```

Comment:
why is this a long long ? you said it should be assigned 1 bit. but this is clearly a big number.